src/django_pg_migration_tools/management/commands/migrate_with_timeouts.py

```diff
@@ -67,6 +188,55 @@ def validate(self) -> None:
                 "At least one of --lock-timeout-in-ms or --statement-timeout-in-ms "
                 "must be specified."
             )
+        self.lock_retry_options.validate()
+
+
+class MigrateRetryStrategy:
+    timeout_options: MigrationTimeoutOptions
+    retries: int
+
+    def __init__(self, timeout_options: MigrationTimeoutOptions):
+        self.timeout_options = timeout_options
+        self.retries = 0
```

Comment:
why not an dataclass.dataclass(frozen=True)?

Reply:
I think there's something strange about putting that much logic behind a dataclass